gravitee-apim-plugin/gravitee-apim-plugin-entrypoint/gravitee-apim-plugin-entrypoint-webhook/src/main/java/io/gravitee/plugin/entrypoint/webhook/WebhookEntrypointConnector.java

```diff
@@ -76,96 +80,101 @@ public int matchCriteriaCount() {
+                    final WebhookEntrypointConnectorConfiguration configuration = connectorFactoryHelper.getConnectorConfiguration(
+                        WebhookEntrypointConnectorConfiguration.class,
+                        subscription.getConfiguration()
+                    );
+
+                    final HttpClientOptions options = new HttpClientOptions();
+                    final String url = configuration.getCallbackUrl();
+                    final URL target = new URL(null, url);
+                    final String protocol = target.getProtocol();
+
+                    if (protocol.charAt(protocol.length() - 1) == 's') {
+                        options.setSsl(true).setUseAlpn(true);
+                    }
+
+                    options.setDefaultHost(target.getHost());
+
+                    if (target.getPort() == -1) {
+                        options.setDefaultPort(options.isSsl() ? 443 : 80);
+                    } else {
+                        options.setDefaultPort(target.getPort());
+                    }
+
+                    final String requestUri = (target.getQuery() == null)
+                        ? target.getPath()
+                        : target.getPath() + URI_QUERY_DELIMITER_CHAR + target.getQuery();
+                    final HttpClient httpClient = vertx.createHttpClient(options);
+
+                    ctx.setInternalAttribute(INTERNAL_ATTR_WEBHOOK_REQUEST_URI, requestUri);
+                    ctx.setInternalAttribute(INTERNAL_ATTR_WEBHOOK_HTTP_CLIENT, httpClient);
+
+                    return Completable.complete();
+                } catch (Exception ex) {
+                    return Completable.error(
+                        new IllegalArgumentException(
+                            "Unable to prepare the HTTP client for the webhook subscription url[" + configuration.getCallbackUrl() + "]",
+                            ex
+                        )
+                    );
+                }
+            }
+        );
+    }
+
+    private Maybe<io.gravitee.gateway.api.buffer.Buffer> sendAndDiscard(String requestUri, HttpClient httpClient, Message message) {
+        // Consume the message in order to send it to the remote webhook and discard it to preserve memory.
+        return httpClient
+            .rxRequest(HttpMethod.POST, requestUri)
+            .flatMap(
+                request -> {
+                    if (message.headers() != null) {
+                        message.headers().forEach(header -> request.putHeader(header.getKey(), header.getValue()));
+                    }
+                    if (message.content() != null) {
+                        return request.rxSend(Buffer.buffer(message.content().getBytes()));
+                    } else {
+                        return request.rxSend();
+                    }
+                }
+            )
+            .flatMapMaybe(response -> Maybe.empty());
```

Comment:
flatMapCompletable ?

Reply:
Yeah you're right it's more readable like this